### ISSUE TITLE
Build will fail on Windows using MSVC due to a new warning introduced in the Build Tools

### ISSUE THREAD

**Issue by jpsouzasilva** [briansmith/ring#658]
### VC++ 2017 version 15.7 v14.14 latest v141 tools.

As said in [this link](https://blogs.msdn.microsoft.com/vcblog/2018/04/20/spectre-diagnostic-in-visual-studio-2017-version-15-7-preview-4/):

`Visual Studio 2017 version 15.7 Preview 4 adds a new capability to our Spectre mitigation: the ability to see where the compiler would have inserted a mitigation and what data led to that action. A new warning, C5045, lets you see what patterns in your code would have caused a mitigation, such as an LFENCE, to be inserted.`

Since the C files are compiled with `/Wall`, the build fails. I could get around this by aliasing the compiler executable to another executable which had the `/wd 5045` flag turned on.

**TL;DR** add the `/wd 5045` flag when compiling with MSVC and it should solve this specific problem.

**Comment by jpsouzasilva** [comment]
@ForsakenHarmony I don't know if it's related or not. It could be similar in the sense of an unwanted warning messing with your build (since warning are treated as errors and unpredictable warnings will make it fail, even if it wouldn't otherwise).

Instead of doing what I did (which is pretty dumb way to get around it), you could instead fork this repo and change the compiler flags yourself until this gets resolved.

**Comment by ForsakenHarmony** [comment]
not entirely sure where that has to be added, why don't you do it? 👀

**Comment by jpsouzasilva** [comment]
@ForsakenHarmony actually, even better, it seems like you can set compiler options through environment variables. I didn't know this when I went through the trouble earlier. 

https://msdn.microsoft.com/en-us/library/kezkeayy.aspx

If the link above isn't working, search for "CL environment variables" on MSDN. I'll test it once I get some time.

**Comment by briansmith** [comment]
Thanks for filing this and thanks to people who posted workarounds. This is fixed in ring-0.13.0-alpha4. I'm closing this as a dupe of #648. See #648 for more details.
